### ISSUE TITLE
Syntax Error while loading YAML.   could not determine a constructor for the tag '!encrypted/pkcs1-oaep'

### ISSUE THREAD

**Issue by aakarshg** [ansible-community/ansible-lint#536]
# Issue Type
- Bug report
- Feature request

# Ansible and Ansible Lint details

```
ansible-lint --version 4.0.1 but can probably be seen in other versions as well
```

Might not be a bug with ansible-lint but, can probably help others who came across a similar issue with ansible-lint failing on files with tags.

The only way to get around having tags in yaml file, and not have ansible-lint throw errors at it is if the file is excluded.

Part of the .yaml file I'd was:

```yaml
- secret:
    name: <some_name>
    data:
      user: <some_user>
      host: github.com
      host_key: <some_value >
      ssh_key: !encrypted/pkcs1-oaep # Issue arises here
```
The error was `Syntax Error while loading YAML.
  could not determine a constructor for the tag '!encrypted/pkcs1-oaep'`

# Workaround

I was using pre-commit here to invoke ansible-lint, so I added the file to exclude list. And it looked as follows.

```
---
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v2.1.0
    hooks:
      - id: mixed-line-ending
      - id: check-byte-order-marker
      - id: check-executables-have-shebangs
      - id: check-merge-conflict
      - id: debug-statements
  - repo: https://github.com/ansible/ansible-lint
    rev: v4.0.1
    hooks:
      - id: ansible-lint
        files: \.(yaml|yml)$
        exclude: ^(some_dir/|\.issue_file\.yaml)
        entry: ansible-lint --force-color -v
```

Just an interesting point to be noted here is that when using pre-commit, pre-commit identifies each modified file and sends it directly to the linter, so adding exclude to the .ansible-lint file doesn't work. Thanks to @ssbarnea who helped me with it.

**Comment by ssbarnea** [comment]
Any updates on this? Apparently is relarted to https://github.com/yaml/pyyaml/issues/294 but I was not able to identify any pyyaml version that can deal with that.

The mentioned workaround does not scale at all. It may buy some time for user having these in 1-3 files but above this it becomes unmaintainable.

**Comment by ssbarnea** [comment]
Also please rename bug title and add a description at at the top that mentioned the error, so it would be easier to find by others:
```
Syntax Error while loading YAML.
  could not determine a constructor for the tag '!encrypted/pkcs1-oaep'
```

**Comment by aakarshg** [comment]
Updated the title to the error description, as suggested.

**Comment by ssbarnea** [comment]
This bug no longer applies because the linter can process generic yaml files that contain the encrypted tag but not as playbooks. When you pass a specific file to the linter it is assumed to be a playbook, and these tags are not valid for playbooks.

Still, if you let the linter detect file types it will not report errors like that, unless you put the file inside a folder named `playbooks/`, as it determines that it must be a playbook.

But as of today, with version v5, if you put a file foo.yml in the root of your repo and put encrypted tags inside it, you will not get an error from the linter.
